probe_scraper/parsers/third_party/parse_histograms.py

```diff
@@ -0,0 +1,1011 @@
+                }
+
+            add_counter("document")
+            add_counter("page")
+
+    return histograms
+
+
+def to_camel_case(property_name):
+    return re.sub(
+        "(^|_|-)([a-z0-9])",
+        lambda m: m.group(2).upper(),
+        property_name.strip("_").strip("-"),
+    )
+
+
+def add_css_property_counters(histograms, property_name):
+    def add_counter(context):
+        name = "USE_COUNTER2_CSS_PROPERTY_%s_%s" % (
+            to_camel_case(property_name),
+            context.upper(),
+        )
+        histograms[name] = {
+            "expires_in_version": "never",
+            "kind": "boolean",
+            "description": "Whether a %s used the CSS property %s"
+            % (context, property_name),
+        }
+
+    add_counter("document")
+    add_counter("page")
+
+
+def from_ServoCSSPropList(filename, strict_type_checks):
+    histograms = collections.OrderedDict()
+    properties = runpy.run_path(filename)["data"]
+    for prop in properties:
+        add_css_property_counters(histograms, prop.name)
+    return histograms
+
+
+def from_counted_unknown_properties(filename, strict_type_checks):
+    histograms = collections.OrderedDict()
+    properties = runpy.run_path(filename)["COUNTED_UNKNOWN_PROPERTIES"]
+
+    # NOTE(emilio): Unlike ServoCSSProperties, `prop` here is just the property
+    # name.
+    #
+    # We use the same naming as CSS properties so that we don't get
+    # discontinuity when we implement or prototype them.
+    for prop in properties:
+        add_css_property_counters(histograms, prop)
+    return histograms
+
+
+# This is only used for probe-scraper.
+def from_properties_db(filename, strict_type_checks):
+    histograms = collections.OrderedDict()
+    with open(filename, "r") as f:
+        in_css_properties = False
+
+        for line in f:
+            if not in_css_properties:
+                if line.startswith("exports.CSS_PROPERTIES = {"):
+                    in_css_properties = True
+                continue
+
+            if line.startswith("};"):
+                break
+
+            if not line.startswith('  "'):
+                continue
+
+            name = line.split('"')[1]
+            add_css_property_counters(histograms, name)
+    return histograms
+
+
+FILENAME_PARSERS = [
+    (lambda x: from_json if x.endswith(".json") else None),
+    (
+        lambda x: from_nsDeprecatedOperationList
+        if x == "nsDeprecatedOperationList.h"
+        else None
+    ),
+    (lambda x: from_ServoCSSPropList if x == "ServoCSSPropList.py" else None),
+    (
+        lambda x: from_counted_unknown_properties
+        if x == "counted_unknown_properties.py"
+        else None
+    ),
+    (lambda x: from_properties_db if x == "properties-db.js" else None),
+]
+
+# Similarly to the dance above with buildconfig, usecounters may not be
+# available, so handle that gracefully.
+try:
+    from . import usecounters
```

Comment:
this is the only line I modified in the vendored parsers, it was originally just `import usecounters`. I'm uncertain whether we should move usecounters to the top level to prevent issues each time this file is updated.

Reply:
I think we're good for patching it like this. I intend to make no future changes to these parsers.